platform-api/src/main/java/io/datacater/core/deployment/DeploymentEndpoint.java

```diff
@@ -1,49 +1,143 @@
 package io.datacater.core.deployment;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
+import io.datacater.core.pipeline.PipelineEntity;
+import io.datacater.core.stream.StreamEntity;
+import io.fabric8.kubernetes.api.model.ListMeta;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
 import io.smallrye.mutiny.Uni;
-import java.util.List;
+import java.util.UUID;
 import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
-import javax.persistence.criteria.CriteriaBuilder;
-import javax.persistence.criteria.CriteriaDelete;
-import javax.persistence.criteria.Root;
 import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
 import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
-import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
 import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
 
 @Path("/api/alpha/deployments")
 @RolesAllowed("dev")
 @Produces(MediaType.APPLICATION_JSON)
-@SecurityRequirement(name = "apiToken")
+@ApplicationScoped
 public class DeploymentEndpoint {
+  private static final Logger LOGGER = Logger.getLogger(DeploymentEndpoint.class);
   @Inject Mutiny.SessionFactory sf;
 
+  @Inject KubernetesClient client;
+
+  @GET
+  @Path("{deploymentName}")
+  public Uni<ObjectMeta> getStream(@PathParam("deploymentName") String deploymentName) {
+    return Uni.createFrom().item(getK8Deployment(deploymentName));
+  }
+
   @GET
-  public Uni<List<DeploymentEntity>> getDeployments() {
-    return sf.withSession(
-        session ->
-            session.createQuery("from DeploymentEntity", DeploymentEntity.class).getResultList());
+  @Path("{deploymentName}/logs")
+  @Produces(MediaType.TEXT_PLAIN)
+  public Uni<String> getLogs(@PathParam("deploymentName") String deploymentName) {
```

Comment:
Can we offer server-sent events for the logs as well? This would make the communication between clients (CLI or UI) and the backend much more efficient.